lib/ts/recipe/session/claimBaseClasses/primitiveClaim.ts

```diff
@@ -0,0 +1,94 @@
+import { JSONPrimitive } from "../../../types";
+import { SessionClaim, SessionClaimValidator } from "../types";
+
+export class PrimitiveClaim<T extends JSONPrimitive> extends SessionClaim<T> {
+    public fetchValue: (userId: string, userContext: any) => Promise<T | undefined> | T | undefined;
+
+    constructor(config: { key: string; fetchValue: SessionClaim<T>["fetchValue"] }) {
+        super(config.key);
+        this.fetchValue = config.fetchValue;
+    }
+
+    addToPayload_internal(payload: any, value: T, _userContext: any): any {
+        return {
+            ...payload,
+            [this.key]: {
+                v: value,
+                t: Date.now(),
+            },
+        };
+    }
+    removeFromPayloadByMerge_internal(payload: any, _userContext?: any): any {
+        const res = {
+            ...payload,
+            [this.key]: null,
+        };
+
+        return res;
+    }
+
+    removeFromPayload(payload: any, _userContext?: any): any {
+        const res = {
+            ...payload,
+        };
+        delete res[this.key];
+
+        return res;
+    }
```

Comment:
if this is not being called anywhere, or if you don't think this is useful for the user either, maybe we should remove this function?

Reply:
Can be used to remove a global claim from the createNewSession override, so we are keeping this.